lib/cdo/user_helpers.rb

```diff
@@ -7,29 +7,39 @@ module UserHelpers
   USERNAME_ALLOWED_CHARACTERS = /[a-z0-9\-\_\.]/
 
   def self.generate_username(queryable, name)
-    prefix = name.downcase.gsub(/[^#{USERNAME_ALLOWED_CHARACTERS.source}]+/, ' ')[0..16].squish.tr(' ', '_')
+    prefix = name.downcase.
+      gsub(/[^#{USERNAME_ALLOWED_CHARACTERS.source}]+/, ' ')[0..16].
+      squish.
+      tr(' ', '_')
 
     if prefix.empty? || prefix == ''
       prefix = 'coder' + (rand(900000) + 100000).to_s
     end
-
     prefix = "coder_#{prefix}" if prefix.length < 5
 
     return prefix if queryable.where(username: prefix).limit(1).empty?
 
+    # Throw darts to find an appropriate suffix, using it if we hit bullseye.
+    (0..1).each do |exponent|
```

Comment:
Four digits was actually my initial implementation (thus the `(0..N).each do |exponent|` loop). My thinking in dropping to one was (a) there is cost to the user to having more digits, (b) there is cost to our DB for the failed queries, and (c) a rare fallback isn't terrible to our DB.

For now, I'm going to leave as is. In a week, when I evaluate whether this actually is a gain, I'll also evaluate whether we want to be falling back more or less often, depending on the timing I see in New Relic.